File: go/international/01_get_quotes.go
package main

import (
	"encoding/json"
	"fmt"
)

func main() {
	baseURL := "https://sandbox.easylink.id"
	appKey := "YOUR_APP_KEY"
	privateKeyPem := "/path/to/private.pem"
	accessToken := "YOUR_ACCESS_TOKEN"

	payload := map[string]interface{}{
		"sourceCurrency": "IDR",
		"targetCurrency": "SGD",
		"amount":         1000000,
	}

	fmt.Println("Fetching Quote...")
	_, res, err := SendEasylinkRequest(baseURL, "/quotes/get-quotes", "POST", payload, appKey, privateKeyPem, accessToken)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
	} else {
		resBytes, _ := json.MarshalIndent(res, "", "  ")
		fmt.Printf("Response:\n%s\n", string(resBytes))
	}
}

File: php/international/01_get_quotes.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/../helpers.php';

// --- Configuration ---
$baseUrl       = 'https://sandbox.easylink.id';
$appId         = 'lQNJ0nL07Ucmemaa';
$appSecret     = 'HrfFeuRmoyBsZhxDi3w3JNdxwYu19lL4';
$appKey        = '3f9a7f74-de23-4fde-af75-da7684528a59';
$privateKeyPem = __DIR__ . '/../../private_key.pem';

// Auto-fetch Access Token
$accessToken   = getAccessToken($baseUrl, $appId, $appSecret);

$payload = [
    'source_currency' => 'IDR',
    'target_currency' => 'SGD',
    'amount'          => 1000000
];

echo "Fetching Quote...\n";
try {
    $res = sendEasylinkRequest(
        $baseUrl,
        '/quotes/get-quotes',
        'POST',
        $payload,
        $appKey,
        $privateKeyPem,
        $accessToken
    );
    echo "Response:\n" . json_encode($res['data'], JSON_PRETTY_PRINT) . "\n";
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}
